### PR TITLE
Arm64Emitter: Convert ShiftType to enum class

### PR DESCRIPTION
**Pull request by Dentomologist**
Straightforward conversion of ShiftType from enum to enum class.

### CODE REVIEW

**Comment by BhaaLseN** [comment]
You can (probably) ignore the Windows fail, thats on our CI...but the Android one needs to be looked at. `m_shifttype` is now a typed enum class and doesn't implicitly convert into `int`, but you can just cast that inside `GetData`.

**Comment by Dentomologist** [comment]
I'm not sure why Visual Studio let that compile in the first place, but I've added the cast.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Review by BhaaLseN — APPROVED** [review]

File: Source/Core/Common/Arm64Emitter.h [hunk 1]
```diff
@@ -277,12 +277,12 @@ constexpr ARM64Reg EncodeRegToQuad(ARM64Reg reg)
...
 }
 
-enum ShiftType
+enum class ShiftType
 {
-  ST_LSL = 0,
-  ST_LSR = 1,
-  ST_ASR = 2,
-  ST_ROR = 3,
+  LogicalShiftLeft = 0,
```
Comment (lioncash) [inline Source/Core/Common/Arm64Emitter.h:282]: I think it'd still be informative as `LSL`, `LSR`, `ASR`, and `ROR` as the names for the shift types. These are what they're referred to as by ARM in official documentation, and the long-form makes code (imo) slightly more verbose to read and write.

Comment (Dentomologist) [reply Source/Core/Common/Arm64Emitter.h]: Given that you've made roughly half the commits touching Arm64Emitter in the last few years, I'm especially inclined to defer to your judgement. I've reverted the naming changes.

Said changes were inspired by my not knowing what the terms referred to until I did some googling. In order to help anyone else in a similar situation I've added some comments in the declaration with the full name, which at least in Visual Studio appear in the hover tooltip for the respective enum values.

**Comment by JosJuice** [comment]
@dolphin-emu-bot rebuild

**Review by JosJuice — APPROVED** [review]

**Review by lioncash — APPROVED** [review]

**Pull request merged**